geth/account/accounts_test.go

```diff
@@ -103,9 +107,373 @@ func TestVerifyAccountPasswordWithAccountBeforeEIP55(t *testing.T) {
 	err = common.ImportTestAccount(keyStoreDir, "test-account3-before-eip55.pk")
 	require.NoError(t, err)
 
-	acctManager := account.NewManager(nil)
+	accManager := NewManager(nil)
 
 	address := gethcommon.HexToAddress(TestConfig.Account3.Address)
-	_, err = acctManager.VerifyAccountPassword(keyStoreDir, address.Hex(), TestConfig.Account3.Password)
+	_, err = accManager.VerifyAccountPassword(keyStoreDir, address.Hex(), TestConfig.Account3.Password)
 	require.NoError(t, err)
 }
+
+var (
+	testErrWhisper    = errors.New("Can't return a whisper service")
+	testErrKeyStore   = errors.New("Can't return a key store")
+	testErrAccManager = errors.New("Can't return an account manager")
+)
+
+func TestManagerTestSuite(t *testing.T) {
+	nodeManager := newMockNodeManager(t)
+	accManager := NewManager(nodeManager)
+
+	keyStoreDir, err := ioutil.TempDir(os.TempDir(), "accounts")
+	require.NoError(t, err)
+	keyStore := keystore.NewKeyStore(keyStoreDir, keystore.LightScryptN, keystore.LightScryptP)
+	defer os.RemoveAll(keyStoreDir) //nolint: errcheck
+
+	testPassword := "test-password"
+
+	// Initial test - create test account
+	nodeManager.EXPECT().AccountKeyStore().Return(keyStore, nil)
+	addr, pubKey, mnemonic, err := accManager.CreateAccount(testPassword)
+	require.NoError(t, err)
+	require.NotEmpty(t, addr)
+	require.NotEmpty(t, pubKey)
+	require.NotEmpty(t, mnemonic)
+
+	s := &ManagerTestSuite{
+		testAccount: testAccount{
+			"test-password",
+			addr,
+			pubKey,
+			mnemonic,
+		},
+		nodeManager:    nodeManager,
+		accManager:     accManager,
+		keyStore:       keyStore,
+		shh:            whisper.New(nil),
+		gethAccManager: accounts.NewManager(),
+	}
+
+	suite.Run(t, s)
+}
+
+func newMockNodeManager(t *testing.T) *common.MockNodeManager {
+	ctrl := gomock.NewController(t)
+	return common.NewMockNodeManager(ctrl)
+}
+
+type ManagerTestSuite struct {
+	suite.Suite
+	testAccount
+	nodeManager    *common.MockNodeManager
+	accManager     *Manager
+	keyStore       *keystore.KeyStore
+	shh            *whisper.Whisper
+	gethAccManager *accounts.Manager
+}
+
+type testAccount struct {
+	password string
+	address  string
+	pubKey   string
+	mnemonic string
+}
+
+// reinitMock is for reassigning a new mock node manager to account manager.
+// Stating the amount of times for mock calls kills the flexibility for
+// development so this is a good workaround to use with EXPECT().Func().AnyTimes()
+func (s *ManagerTestSuite) reinitMock() {
+	s.nodeManager = newMockNodeManager(s.T())
```

Comment:
Only twp lines. So move into `SetupTest()` (including comment).

Reply:
I mean, `reinitMock` is also called multiple times in test functions. It's mainly for dropping all calls whenever needed and use the mock as an ever-changing stub for different cases in a test function, especially during table-driven tests. That's why I objected to "extracting" :)

Since it's also called in the beginning of every test function, it sounded reasonable to use `SetupTest`, though.

Reply:
Ah, okay, missed it. So it's fine.